aws-lambda-java-runtime-interface-client/src/main/jni/com_amazonaws_services_lambda_runtime_api_client_runtimeapi_NativeClient.cpp

```diff
@@ -56,50 +95,40 @@ JNIEXPORT jobject JNICALL Java_com_amazonaws_services_lambda_runtime_api_client_
     return NULL;
   }
 
-  jclass invocationRequestClass;
-  jfieldID invokedFunctionArnField;
-  jfieldID deadlineTimeInMsField;
-  jfieldID xrayTraceIdField;
-  jfieldID idField; 
   jobject invocationRequest;
-  jfieldID clientContextField;
-  jfieldID cognitoIdentityField;
   jbyteArray jArray;
   const jbyte* bytes;
-  jfieldID contentField;
   auto response = outcome.get_result();
 
-  CHECK_EXCEPTION(env, invocationRequestClass = env->FindClass("com/amazonaws/services/lambda/runtime/api/client/runtimeapi/InvocationRequest"));
   CHECK_EXCEPTION(env, invocationRequest = env->AllocObject(invocationRequestClass));
-
-  CHECK_EXCEPTION(env, idField = env->GetFieldID(invocationRequestClass , "id", "Ljava/lang/String;"));
   CHECK_EXCEPTION(env, env->SetObjectField(invocationRequest, idField, env->NewStringUTF(response.request_id.c_str())));
-  
-  CHECK_EXCEPTION(env, invokedFunctionArnField = env->GetFieldID(invocationRequestClass , "invokedFunctionArn", "Ljava/lang/String;"));
   CHECK_EXCEPTION(env, env->SetObjectField(invocationRequest, invokedFunctionArnField, env->NewStringUTF(response.function_arn.c_str())));
-
-  CHECK_EXCEPTION(env, deadlineTimeInMsField = env->GetFieldID(invocationRequestClass , "deadlineTimeInMs", "J"));
   CHECK_EXCEPTION(env, env->SetLongField(invocationRequest, deadlineTimeInMsField, std::chrono::duration_cast<std::chrono::milliseconds>(response.deadline.time_since_epoch()).count()));
 
   if(response.xray_trace_id != ""){
-    CHECK_EXCEPTION(env, xrayTraceIdField = env->GetFieldID(invocationRequestClass , "xrayTraceId", "Ljava/lang/String;"));
+    if(xrayTraceIdField == nullptr){
```

Comment:
why are these conditions needed for `xrayTraceIdField`, `clientContextField`, `cognitoIdentityField`?

Reply:
@smirnoal Not all next responses contain them, so it would be more efficient to load them lazily than eagerly. Note that they are still cached once they are loaded.

Reply:
I'd  move them to `JNI_OnLoad` to have all the fields initialisation in one place and reduce complexity here a bit. It would be good to know performance implications of this, I'm just assuming they are negligible